cli/JSXV4.md

```diff
@@ -76,15 +76,95 @@ module Wrapper = {
 let make = () => body
 ```
 
-**V4 Spec**
+### Migration of V3 components that depend on the internal representation
+
+Some components in existing projects are written in a way that is dependent on the V3 internal representation.
+Here are a few examples of how to convert them to V4.
+
+#### `makeProps` does not exist in V4
+
+Rewrite this:
+
+```rescript
+// V3
+module M = {
+  @obj external makeProps: (~msg: 'msg, ~key: string=?, unit) => {"msg": 'msg} = "" // No more makeProps
+
+  let make = (~msg) => {
+    <div> {React.string(msg)} </div>
+  }
+}
+```
+
+To this:
+
+```rescript
+// V4
+module M = {
+  type props<'msg> = {msg: 'msg}
+  let make = props => <div> {React.string(props.msg)} </div>
+}
+```
+
+#### `React.Context`
+
+```rescript
+module Context = {
+  let context = React.createContext(() => ())
+
+  module Provider = {
+    let provider = React.Context.provider(context)
+
+    @react.component
+    let make = (~value, ~children) => {
+      React.createElement(provider, {"value": value, "children": children}) // Error
+      // Fix : use the record
+      React.createElement(provider, {value, children}) // OK
```

Comment:
Expand the example seperately? Thought?

Reply:
Same goes here https://github.com/rescript-lang/syntax/pull/652/commits/22ec394d56b800b3446af0f6a33334fa4625f6f8